packages/perseus-editor/src/components/__tests__/locked-function-settings.test.tsx

```diff
@@ -0,0 +1,365 @@
+            // Arrange
+            const onChangeProps = jest.fn();
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the axis direction
+            const directionalAxisSwitch =
+                screen.getByLabelText("equation prefix");
+            await userEvent.click(directionalAxisSwitch);
+            const axisOption = screen.getByText("x =");
+            await userEvent.click(axisOption);
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledWith({directionalAxis: "y"});
+        });
+
+        describe("Domain restrictions", () => {
+            test("valid entries update component properties", async () => {
+                // Arrange
+                render(
+                    <LockedFunctionSettings
+                        {...defaultProps}
+                        expanded={true}
+                        onChangeProps={onChangeProps}
+                    />,
+                    {wrapper: RenderStateRoot},
+                );
+
+                // Act
+                const domainMinField = screen.getByLabelText("domain min");
+                await userEvent.type(domainMinField, "1");
+                const domainMaxField = screen.getByLabelText("domain max");
+                await userEvent.type(domainMaxField, "5");
+
+                // Assert
+                expect(onChangeProps).toHaveBeenCalledTimes(2);
+                expect(onChangeProps).toHaveBeenNthCalledWith(1, {
+                    domain: [1, Infinity],
+                });
+                expect(onChangeProps).toHaveBeenNthCalledWith(2, {
+                    domain: [-Infinity, 5],
+                });
+            });
+
+            test("negative entries are handled appropriately", async () => {
+                // Arrange
+                render(
+                    <LockedFunctionSettings
+                        {...defaultProps}
+                        expanded={true}
+                        onChangeProps={onChangeProps}
+                    />,
+                    {wrapper: RenderStateRoot},
+                );
+
+                // Act
+                const domainMinField = screen.getByLabelText("domain min");
+                await userEvent.type(domainMinField, "-5");
+                await userEvent.type(domainMinField, "[Backspace/]4");
+                await userEvent.type(
+                    domainMinField,
+                    "[Backspace/][Backspace/]-3",
+                );
+                const domainMaxField = screen.getByLabelText("domain max");
+                await userEvent.type(domainMaxField, "-2");
+
+                // Assert
+                // expect(onChangeProps).toHaveBeenCalledTimes(2);
```

Comment:
I suspect you didn't mean to leave this comment here

Reply:
Correct. Good catch. Thank you.